onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -1056,6 +1056,29 @@ def testSharedAllocatorUsingCreateAndRegisterAllocator(self):
         so2.log_severity_level = 1
         onnxrt.InferenceSession(get_name("mul_1.onnx"), sess_options=so2, providers=onnxrt.get_available_providers())
 
+    def testMemoryArenaShrinkage(self):
+        x = np.array([[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]], dtype=np.float32)
+
+        sess1 = onnxrt.InferenceSession(get_name("mul_1.onnx"), providers=['CPUExecutionProvider'])
+        input_name = sess1.get_inputs()[0].name
+
+        # Shrink CPU memory after execution
+        ro1 = onnxrt.RunOptions()
+        ro1.add_run_config_entry("memory.enable_memory_arena_shrinkage", "cpu:0")
+        self.assertEqual(ro1.get_run_config_entry("memory.enable_memory_arena_shrinkage"), "cpu:0")
```

Comment:
The x86 test failure is because on x86 builds, the CPU allocator does not use an arena and it can't honor the shrinkage request - https://github.com/microsoft/onnxruntime/blob/d478a53d433473cad296bcdae2edc43a0fef210c/onnxruntime/core/providers/cpu/cpu_execution_provider.h#L37

Can you please adjust the test to account for that ?

Reply:
Changed to skip tests on x86 builds.
It will probably work correctly, but I do not have a 32bit windows environment.
Could you please check with CI?

Reply:
Sure, thanks again